spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-polling-receiver/src/main/java/com/example/PollingReceiverApplication.java

```diff
@@ -46,11 +47,16 @@ public static void main(String[] args) {
     SpringApplication.run(PollingReceiverApplication.class, args);
   }
 
+  @Bean
+  public String subscription() {
+    return "exampleSubscription";
```

Comment:
Have you considered just using a property and the `@Value` annotation to inject it?

Reply:
That's a nice idea. Tried this out in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/2085/commits/919c081010d29ec68ac938b37f312e327484fcf3. However, the problem with using `@Value` and `${random.uuid}` in application.properties is that the value doesn't stay consistent and the random uuid get regenerated every time the `@Value` annotation is used. In our particular case, the subscriptionName is `subscription-name-51bb5af5-354b-4b03-9aa9-297b31b30b88` in the TestConfiguration class and `subscription-name-12f7f591-7955-4453-98b3-2f91372c1ef6` in the PollerReceiverApplication, therefore resulting our test failing with the RESOURCE_NOT_FOUND error.

Reply:
That makes sense, but can you just use `@Value` once for `subscriptionName`?

Reply:
Done, tested with running integration test and sample locally.

Reply:
I was thinking more along these lines: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/2103

Reply:
Ah thank you! Sorry, I thought we wanted to avoid using a bean all together.  The example really helped! Made the change.